backend/src/database/users.go

```diff
@@ -35,6 +40,31 @@ func (db *DB) GetCurrentUsers(page, itemsPerPage int, facilityId uint, order str
 	return count, users, nil
 }
 
+func (db *DB) GetAdminOrStudentUsers(page, itemsPerPage int, facilityId uint, order string, search string, role string) (int64, []models.User, error) {
```

Comment:
instead of duplicating this function, can we instead just add a check for the query like
```go
// also make sure it's a pointer to models.User
tx := db.Modal(&models.User{}).("facility_id = ?", facilityId)

switch role {
case "admin":
tx = tx.Where("role = 'admin'")
case "student":
tx = tx.Where("role = 'student'")
}
```
that way you can keep all this in one function :+1: